pylearn2/train.py

```diff
@@ -140,26 +140,34 @@ def main_loop(self, time_budget=None):
         t0 = datetime.now()
         self.setup()
         if self.algorithm is None:
-            self.run_callbacks_and_monitoring()
-            while True:
-                if self.exceeded_time_budget(t0, time_budget):
-                    break
+            extension_continue = self.run_callbacks_and_monitoring()
+            # First check if the model is already beyond the stop criteria of
+            # training, if so, just return directly.
+            continue_learning = (self.model.continue_learning() and
+                                 extension_continue)
+            assert continue_learning in [True, False, 0, 1]
+            if continue_learning:
+                while True:
```

Comment:
I think it would be cleaner to just have `while continue_learning:` here and drop the `if not continue_learning: break` at the end of the loop.